aspnetcore/mvc/views/view-compilation.md

```diff
@@ -1,66 +1,92 @@
+A Razor file is compiled at runtime, when the associated Razor Page or MVC view is invoked. Razor files are compiled at both build and publish time using the [Razor SDK](xref:mvc/razor-pages/sdk).
+::: moniker-end
 
+## Precompilation considerations
 
+The following are side effects of precompiling Razor files:
 
-Precompilation considerations:
+* A smaller published bundle
+* A faster startup time
+* You can't edit Razor files&mdash;the associated content is absent from the published bundle.
 
-* Precompiling views results in a smaller published bundle and faster startup time.
-* You can't edit Razor files after you precompile views. The edited views won't be present in the published bundle. 
+## Deploy precompiled files
 
-To deploy precompiled views:
+::: moniker range=">= aspnetcore-2.1"
+Build- and publish-time compilation of Razor files is enabled by default by the Razor SDK. Editing Razor files after they're updated is supported at build time. By default, only the compiled *Views.dll* and no *.cshtml* files are deployed with your app.
 
-# [ASP.NET Core 2.1](#tab/aspnetcore21/)
-Build and publish time compilation of Razor files is enabled by default by the Razor Sdk. Editing Razor files after they are updated is supported at build time. By default, only the compiled *Views.dll* and no cshtml files are deployed with your application. 
-    
 > [!IMPORTANT]
-> The Razor Sdk is only effective when no precompilation specific properties are set in your project file. For instance, setting `MvcRazorCompileOnPublish` in your *.csproj* file will disable the Razor Sdk.
+> The Razor SDK is effective only when no precompilation-specific properties are set in the project file. For instance, setting the *.csproj* file's `MvcRazorCompileOnPublish` property to `true` disables the Razor SDK.
+::: moniker-end
 
-# [ASP.NET Core 2.0](#tab/aspnetcore20/)
+::: moniker range="= aspnetcore-2.0"
+If your project targets .NET Framework, install the [Microsoft.AspNetCore.Mvc.Razor.ViewCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/) NuGet package:
 
-If your project targets .NET Framework, include a package reference to [Microsoft.AspNetCore.Mvc.Razor.ViewCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/):
-
-```xml
-<PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.ViewCompilation" Version="2.0.0" PrivateAssets="All" />
-```
+[!code-xml[](view-compilation/sample/DotNetFrameworkProject.csproj?name=snippet_ViewCompilationPackage)]
 
 If your project targets .NET Core, no changes are necessary.
 
-The ASP.NET Core 2.x project templates implicitly sets `MvcRazorCompileOnPublish` to `true` by default, which means this node can be safely removed from the *.csproj* file.
-    
+The ASP.NET Core 2.x project templates implicitly set the `MvcRazorCompileOnPublish` property to `true` by default. Consequently, this element can be safely removed from the *.csproj* file.
```

Comment:
Should we say we recommend removing the element from your csproj if it is set to `true`? Having the property will turn off Razor Sdk in 2.1, so it's less than desirable